go/vt/discovery/endpointwatcher_test.go

```diff
@@ -0,0 +1,136 @@
+package discovery
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	pbt "github.com/youtube/vitess/go/vt/proto/topodata"
+	"github.com/youtube/vitess/go/vt/topo"
+	"github.com/youtube/vitess/go/vt/topo/test/faketopo"
+	"golang.org/x/net/context"
+)
+
+func TestEndPointWatcher(t *testing.T) {
+	ft := newFakeTopo()
+	fhc := newFakeHealthCheck()
+	t.Logf(`ft = FakeTopo(); fhc = FakeHealthCheck()`)
+	epw := NewEndPointWatcher(topo.Server{Impl: ft}, fhc, []string{"aa"}, 10*time.Millisecond)
+	t.Logf(`epw = EndPointWatcher(topo.Server{ft}, fhc, ["aa"], 10ms)`)
+
+	ft.AddTablet("aa", 0, "host1", map[string]int32{"vt": 123})
+	time.Sleep(100 * time.Millisecond)
```

Comment:
sleeps are bad in these tests, you're flaky for sure.

Above, I suggested to break the watch timer and the 'watch' method into two functions, if you do that, the added benefit is in here: you can call the internal function without depending on the timer ever.